### PR TITLE
Feature/beta channel

### PR DESCRIPTION
**Pull request by yonip23**

### CODE REVIEW

File: src/preRelease/versions.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export function getAvailableAlphaVersion(artifactUrl: string): string {
+  const versionPattern = /(?<=download\/)(.*)(?=\/tabnine-vscode)/gi;
+  const match = artifactUrl.match(versionPattern);
+  return (match && match[0]) || "";
+}
+
+export function userConsumesAlphaVersions(): boolean {
+  const isVersionSupported = semver.gte(
```
Comment (dimacodota) [inline src/preRelease/versions.ts:37]: you need to check this for beta as well

**Review by dimacodota — APPROVED** [review]
see my comment

File: src/globals/tabnineExtensionProperties.ts [hunk 2]
```diff
@@ -59,6 +61,12 @@ function getContext(): TabNineExtensionProperties {
       true
     );
   }
+  const isExtentionBetaChannelEnabled =
+    configuration.get<boolean>("tabnine.receiveBetaChannelUpdates") || false;
+
+  const isVscodeInsiders =
+    vscode.env.appName.toLocaleLowerCase().indexOf("insider") > -1;
```
Comment (dimacodota) [inline src/globals/tabnineExtensionProperties.ts:68]: you can use .includes

File: src/preRelease/versions.ts (hunk 1 above)
Comment (yonip23) [reply src/preRelease/versions.ts:37]: Why is this needed? What isn't supported on previous versions?

**Pull request merged**